cypress/integration/dashboard/dashboard_spec.js

```diff
@@ -1,59 +1,147 @@
+    const slug = Cypress._.get(response, 'body.slug');
+    assert.isDefined(slug, 'Widget api call returns widget slug');
     return slug;
   });
 }
 
-function archiveCurrentDashboard() {
+function editDashboard() {
   cy.getByTestId('DashboardMoreMenu')
     .click()
     .within(() => {
       cy.get('li')
-        .contains('Archive')
+        .contains('Edit')
         .click();
     });
+}
 
-  cy.get('.btn-warning')
-    .contains('Archive')
-    .click();
-  cy.get('.label-tag-archived').should('exist');
+function addTextbox(text) {
+  cy.server();
+  cy.route('POST', 'api/widgets').as('NewWidget');
+
+  editDashboard();
+
+  cy.contains('a', 'Add Textbox').click();
+  cy.get('.add-textbox').within(() => {
+    cy.get('textarea').type(text);
+  });
+  cy.contains('button', 'Add to Dashboard').click();
+  cy.get('.add-textbox').should('not.exist');
+  cy.contains('button', 'Apply Changes').click();
+
+  return cy.wait('@NewWidget').then((xhr) => {
+    const id = Cypress._.get(xhr, 'response.body.id');
+    assert.isDefined(id, 'Widget api call returns widget id');
+    return cy.getByTestId(`WidgetId${id}`).should('exist');
+  });
 }
 
 describe('Dashboard', () => {
   beforeEach(() => {
     cy.login();
   });
 
-  it('creates a new dashboard and archives it', () => {
-    createNewDashboard('Foo Bar').then((slug) => {
+  it('creates new dashboard', () => {
+    cy.visit('/dashboards');
+    cy.getByTestId('CreateButton').click();
+    cy.get('li[role="menuitem"]').contains('Dashboard').click();
+
+    cy.server();
+    cy.route('POST', 'api/dashboards').as('NewDashboard');
+
+    cy.getByTestId('EditDashboardDialog').within(() => {
+      cy.getByTestId('DashboardSaveButton').should('be.disabled');
+      cy.get('input').type('Foo Bar');
+      cy.getByTestId('DashboardSaveButton').click();
+    });
+
+    cy.wait('@NewDashboard').then((xhr) => {
+      const slug = Cypress._.get(xhr, 'response.body.slug');
+      assert.isDefined(slug, 'Dashboard api call returns slug');
+
       cy.visit('/dashboards');
       cy.getByTestId('DashboardLayoutContent').within(() => {
-        cy.getByTestId(slug).should('exist').click();
+        cy.getByTestId(slug).should('exist');
+      });
+    });
+  });
+
+  it('archives dashboard', function() {
+    createNewDashboardByAPI('Foo Bar').then((slug) => {
+      cy.visit(`/dashboard/${slug}`);
+
+      cy.getByTestId('DashboardMoreMenu')
+      .click()
+      .within(() => {
+        cy.get('li')
+          .contains('Archive')
+          .click();
       });
 
-      archiveCurrentDashboard();
+      cy.get('.btn-warning')
+        .contains('Archive')
+        .click();
+      cy.get('.label-tag-archived').should('exist');
 
       cy.visit('/dashboards');
       cy.getByTestId('DashboardLayoutContent').within(() => {
         cy.getByTestId(slug).should('not.exist');
       });
     });
   });
+
+
+  describe('Editing a dashboard', () => {
```

Comment:
:thinking:. I actually thought about the `context` where you'd be "editing a dashboard". But thinking better you can also use `describe` in this case. A good pattern for this:
- Use `context` when you are testing the same thing, but within a context - `context('editing a dashboard', () =>{})`;
- Use `describe` when you are testing a different thing :slightly_smiling_face: - `describe('Textbox', () =>{})` 

In the `context` case I'd also add the `editDashboard()` to the `beforeEach` (we are in this context, so it makes sense).

For this it's really open, you can either consider one of this options, or even aggregate both :man_shrugging:

Reply:
Sorry about suggesting the exact thing you used before :sweat_smile:, but if you choose the `describe` option:
```suggestion
  describe('Textbox', () => {
```

In my defense I haven't thought we'd ever use aggregated `describe`'s, but turns out it makes sense :stuck_out_tongue_closed_eyes: